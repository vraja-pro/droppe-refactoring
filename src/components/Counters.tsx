import React,{FC} from 'react'
import styles from "../shopApp.module.css";

type Props = {
    total:number,
    favorites:number
}

const Counters:FC<Props> = ({total,favorites}) => {
  return (
    <div className={styles.statsContainer}>
    <span>{`Total products: ${total} - Number of favorites: ${favorites}`}</span>
 </div>
  )
}

export default Counters