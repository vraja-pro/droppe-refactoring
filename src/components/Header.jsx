import React from 'react'

type Props = {}

const Header = (props: Props) => {
  return (
    <div className={styles.header}>
    <div className={['container', styles.headerImageWrapper].join(' ')}>
      <img src={logo} className={styles.headerImage} />
    </div>
  </div>
  )
}

export default Header