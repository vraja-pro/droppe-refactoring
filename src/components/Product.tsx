import React,{FC} from 'react'
import {product} from '../interfaces'
import styles from "./product-list-components.module.css";
import { FaStar } from "react-icons/fa";

interface Props extends product{
    onFav:(id:string)=>void;
}

const Product:FC<Props> = ({title,rating,price,description,id,onFav,isFavorite}) => {
    const {product: productClass, productBody, actionBarItem, actionBarItemLabel,action_bar,} = styles
  return (
    <div className={productClass}>
        <h3 className={styles['product-title']}>{title}</h3>

    <p><b>Rating: {rating ? `${rating.rate}/5` : ''}</b></p>

    <p><b>Price: ${price}</b></p>

    <p className={productBody}>
      <span><b>Description:</b></span>
      <br/>
      {description}
   </p>

    <div className={`${action_bar}`}>
      <button
        className={`${actionBarItem} ${isFavorite ? "active" : ""}`}
        onClick={() => {onFav(id)}}>
        <FaStar /> <span className={actionBarItemLabel}>{isFavorite ? 'Remove from favorites' : 'Add to favorites'}</span>
      </button>
    </div>
  
  </div>
  )
}

export default Product