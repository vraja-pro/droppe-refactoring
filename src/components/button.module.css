.button {
  border: 0;
  cursor: pointer;
  color: #ffffff;
  font-size: 18px;
  background-color: #62bfad;
  border-radius: 10px;
  width: 100%;
  padding: 12px;
}

.button:focus {
  outline: none;
  margin: inherit;
  border: inherit;
  -webkit-align-content: inherit;
  -webkit-animation-delay: inherit;
  -webkit-animation-fill-mode: inherit;
  -webkit-animation-timing-function: inherit;
  -webkit-box-decoration-break: inherit;
}
