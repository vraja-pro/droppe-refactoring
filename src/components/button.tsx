import React,{FC} from "react";
import styles from "./button.module.css";
import {ButtonProps} from '../interfaces'

export const Button:FC<ButtonProps> = ({ children, onClick }) => (
  <button className={styles.button} onClick={onClick}>
    {children}
  </button>
);
