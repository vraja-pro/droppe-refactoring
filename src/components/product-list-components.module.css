.product {
  color: #0f2137;
  background-color: #ffffff;
  border: 1px solid #e3e4e8;
  border-radius: calc(1.5 * 4px);
  width: 100%;
  height: auto;
  padding: 16px 24px 8px;
  margin: inherit;
}

.product:not(:last-child) {
  margin-bottom: 16px;
}

.product-title {
  font-size: 24px;
  font-weight: 600;
  margin: 0 0 16px;
}

.productBody {
  margin: 0 0 16px;
}

.action_bar {
  display: flex;
  justify-content: flex-start;
  border-top: 1px solid #e3e4ee;
  padding-top: 8px;
}

button.actionBarItem {
  cursor: pointer;
  border-radius: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 8px;
  width:100%;
  outline: none;
  border:none;
  background:none;
  font-size: 16px;
}

button.actionBarItem:hover {
  background-color: #f4f5f9;
}

button.actionBarItem:global(.active) {
  color: #62bfad;
}

.actionBarItemLabel {
  margin-left: 8px;
}
