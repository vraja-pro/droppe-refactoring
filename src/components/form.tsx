import React,{FC,FormEvent} from "react";
import { Button } from "./button";
import styles from "./form.module.css";
import {ProductFormElement,FormProps} from '../interfaces'

export const Form: FC<FormProps> = ({onSubmit}) => {

  const handleSubmit = (e:FormEvent<ProductFormElement>) => {
    
    e.preventDefault();
    if (!e.currentTarget.elements.title.value) {
      alert("Your product needs a title");
      return;
    }

    if (!e.currentTarget.elements.description.value || !e.currentTarget.elements.price.value) {
      alert("Your product needs some content");
      return;
    }
    onSubmit({
      title: e.currentTarget.elements.title.value,
      description: e.currentTarget.elements.description.value,
      price: e.currentTarget.elements.price.value,
    });

    e.currentTarget.reset();
  };

  return (
    <form className={styles.form} onSubmit={handleSubmit} data-testid="add-product-form">
      <label className={styles.label}>Product title: *</label>

      <input
        name="title"
        placeholder="Title..."
        className={styles.input}
      />

      <label className={styles.label}>Product details: *</label>

      <input
        name="price"
        placeholder="Price..."
        className={styles.input}
      />

      <textarea
        name="description"
        placeholder="Start typing product description here..."
        className={styles.textarea}
      />

      <Button>Add a product</Button>
    </form>
  );
};
