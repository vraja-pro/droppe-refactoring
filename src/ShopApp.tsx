import React, {useState, useEffect} from "react";
import lodash from 'lodash';
import Modal from "react-modal";
import { FaTimes } from "react-icons/fa";
import { Button } from "./components/button";
import ProductList from "./components/product-list-components";
import { Form } from "./components/form";
import {PRODUCTS_API} from "./constants";
import Header from './components/Header';
import axios from "axios";
import Product from './components/Product';


type Rating = {
  rate:number;
}
export interface product {
  id:string,
  title:string;
  description:string;
  price: string;
  isFavorite?:boolean;
  rating?:Rating;
}

type data = {
  products: product[]; 
  isOpen: boolean; 
  isShowingMessage: boolean;
  message:string; 
  numFavorites: number; 
  prodCount: number;
}

 export const ShopApp = () => {
   const [shopData,setShopData] = useState<data>({products:[], isOpen: false, isShowingMessage: false, message: '', numFavorites: 0, prodCount: 0 });
  const [products,setProducts] = useState<product[]>([])

   const getProducts = () => {
      axios(PRODUCTS_API).then(((response)=>{
        setProducts(response.data)
      })).catch(err=>{
        console.log(err)
      }); 
   }

   const favClick = (id:string) => {
     const updatedProducts = products.map((obj:product)=>{
      if(obj.id == id){
          return {...obj,isFavorite:obj?.isFavorite ? !obj.isFavorite : true};
      }else{
        return obj;
      }
    })
    setProducts(updatedProducts)
    //console.log(updatedProducts)
   }

   useEffect(  () => {
   getProducts();
    
   }, [])
   

    return (
      <>
      
         <Header /> 
      
      <div className="container">
       
         {products && products.map((obj)=><Product {...obj} onFav={favClick} key={obj.id}/>)}
         </div>
           
        
      </>
    );
  
}
