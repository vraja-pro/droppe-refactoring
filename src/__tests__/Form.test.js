import React from 'react'
import {render,screen, cleanup, fireEvent} from '@testing-library/react'
import {Form} from '../components/form';
import ShopApp from '../ShopApp';
import renderer from 'react-test-renderer'

afterEach(()=>{
    cleanup();
});
test('form matches snapshot', () => { 
    const onSubmit = jest.fn;
   const tree = renderer.create(<Form onSubmit={onSubmit}/>).toJSON();
    expect(tree).toMatchSnapshot();
 })
test('is form being rendered', () => { 
    
    const showModal = jest.fn;
    render(<ShopApp />);
    const button = screen.getByText('Send product proposal');
    fireEvent.click(button)
    expect(showModal).toHaveBeenCalled;
    const productFormElement = screen.getByTestId('add-product-form');
    expect(productFormElement).toBeInTheDocument();
   
})

