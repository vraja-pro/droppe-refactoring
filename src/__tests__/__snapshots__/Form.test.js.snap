// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form matches snapshot 1`] = `
<form
  className="form"
  data-testid="add-product-form"
  onSubmit={[Function]}
>
  <label
    className="label"
  >
    Product title: *
  </label>
  <input
    className="input"
    data-testid="title"
    name="title"
    placeholder="Title..."
  />
  <label
    className="label"
    htmlFor="description"
  >
    Product details: *
  </label>
  <input
    className="input"
    data-testid="price"
    name="price"
    placeholder="Price..."
  />
  <textarea
    className="textarea"
    data-testid="description"
    name="description"
    placeholder="Start typing product description here..."
  />
  <button
    className="button"
  >
    Add a product
  </button>
</form>
`;
